crates/l2/docs/withdrawals.md

```diff
@@ -0,0 +1,23 @@
+# Withdrawal specs
+
+This document contains a detailed explanation of the changes needed to handled withdrawals and the withdrawal flow.
+
+A new `Withdraw` type of transaction on L2 is introduced, where users send a certain amount of `eth` (or the native token in a custom token setup) with it. This money is then burned on the L2 and the operator sends a `WithdrawLog` to L1, so the user can then send a transaction to claim the withdrawal associated to that log and receive their funds from the Common bridge.
+
+In more detail, the full changes/additions are:
+
+- A `Withdraw` transaction type is introduced, comprised of the regular fields in an EIP-1559 transaction.
+- On every block, each `Withdraw` transaction will burn (i.e. deduct from the sender) the value attached to it. Withdraw transactions will also emit a `WithdrawLog` coming from a privileged address (for now the zero address). This log will contain the following information:
+    - `to`: the address in L1 that is allowed to claim the funds (this is decided by the user as part of a withdraw transaction. This comes from the regular `to` field on the Withdraw transaction (i.e. we are reusing that field with a slightly different meaning; what it means here is “the address that can claim the funds on L1”).
+    - `amount`: the amount of money withdrawn (i.e. the `msg.value` of the transaction).
+    - `tx_index`: the index of the transaction in the L2 block it was included in. This will be important for claiming the withdrawal as it will require a merkle proof to be provided along with the index on the tree.
```

Comment:
```suggestion
    - `tx_hash`: the transaction hash in the L2 block it was included in. This will be important for claiming the withdrawal as it will require a merkle proof to be provided along with the index on the tree.
```